src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
This is a really good catch Scott.  This is exactly the kind of stuff I'm trying to clean up.  I'm exploring if I can make this `const`.  It leads to marking a lot of other members `const` and I'm still trying to get to the bottom of the rabbit hole...